cardano-node/src/Cardano/Node/Tracers.hs

```diff
@@ -132,6 +80,104 @@ data Tracers peer blk = Tracers {
 
 -- | Smart constructor of 'NodeTraces'.
 --
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => Transformable Text IO (WithTip blk (ChainDB.TraceEvent blk)) where
+    trTransformer = trStructured -- structure required, will call 'toObject'
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => ToObject (WithTip blk (ChainDB.TraceEvent blk)) where
+    toObject MinimalVerbosity _ = emptyObject -- no output
+    toObject verb (WithTip tip ev) =
+        mkObject [ "tip" .= showTip tip
+                 , "TraceEvent" .= toObject verb ev
+                 ]
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => ToObject (ChainDB.TraceEvent blk) where
+    toObject MinimalVerbosity _ = emptyObject -- no output
+    toObject _ (ChainDB.TraceAddBlockEvent ev) =
+        mkObject [ "kind" .= String "TraceAddBlockEvent"
+                 , "event" .= String (
+                                pack (
+                                  case ev of
+                                    ChainDB.StoreButDontChange   pt ->
+                                      "Ignoring block: " <> condense pt
+                                    ChainDB.TryAddToCurrentChain pt ->
+                                      "Block fits onto the current chain: " <> condense pt
+                                    ChainDB.TrySwitchToAFork pt _   ->
+                                      "Block fits onto some fork: " <> condense pt
+                                    ChainDB.SwitchedToChain _ c     ->
+                                      "Chain changed, new tip: " <> condense (AF.headPoint c)
+                                    ChainDB.AddBlockValidation ev'  -> case ev' of
+                                      ChainDB.InvalidBlock err pt
+                                        -> "Invalid block " <> condense pt <> ": " <> show err
+                                      _ -> "ignore"
```

Comment:
I don't think we want to be tracing `"ignore"` here

Reply:
yes, these should become `emptyObject` as you proposed below.